mqclient/broker_clients/rabbitmq.py

```diff
@@ -340,9 +344,12 @@ async def message_generator(
                 else:
                     pass
 
-        # Garbage Collection (or explicit close(), or break in consumer's loop)
+            self.channel.cancel()
+
+        # Garbage Collection (or explicit generator close(), or break in consumer's loop)
         except GeneratorExit:
             LOGGER.debug(log_msgs.MSGGEN_GENERATOR_EXITING)
+            self.channel.cancel()
```

Comment:
If cancel is both before and after the `except`, should it maybe be in `finally` instead?

Reply:
Good point